uaclient/api/u/pro/attach/magic/wait/v1.py

```diff
@@ -54,6 +54,7 @@ def __init__(
 def wait(
     options: MagicAttachWaitOptions,
 ) -> MagicAttachWaitResult:
+    secret_manager.secrets.add_secret(options.magic_token)
```

Comment:
please move this to the top of the `_wait` function (we use that one directly in internal calls)